code/crates/sync/src/handle.rs

```diff
@@ -243,8 +232,6 @@ where
 {
     debug!(%response.height, %request_id, %peer_id, "Received response");
 
-    state.remove_pending_value_request_by_height(&response.height);
-
     let response_time = metrics.value_response_received(response.height.as_u64());
 
```

Comment:
>  So sync should be able to handle the case where it sends the request then sees decided from consensus and then response.

Then shouldn't we check here that `pending_value_requests` is different than `RequestState::Validated`? If consensus already decided on a value and changed the status in `on_decided`, then `on_value_response` should not call `request_value_from_peer_except`.